pkgs/by-name/ne/networkmanager-ssh/package.nix

```
@@ -0,0 +1,65 @@
+{
+  stdenv,
+  lib,
+  autoreconfHook,
+  fetchFromGitHub,
+  pkg-config,
+  networkmanager,
+  libsecret,
+  gtk3,
+  gtk4,
+  gnome,
+  glib,
+  libnma,
+  libnma-gtk4,
+  intltool,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "NetworkManager-ssh";
+  version = "1.4.2";
+
+  src = fetchFromGitHub {
+    owner = "danfruehauf";
+    repo = "NetworkManager-ssh";
+    rev = "tags/${version}";
+    sha256 = "sha256-ExCU22V4fYuFXW/HqJ39+PbYykXu4rpk8+3/hg9KTMo=";
+  };
+
+  patches = [
+    ./fix-paths.patch
+  ];
+
+  nativeBuildInputs = [
+    autoreconfHook
+    pkg-config
+    intltool
+    gtk4
+  ];
+
+  buildInputs = [
+    networkmanager
+    glib
+    gtk3
+    gtk4
+    libsecret
+    libnma
+    libnma-gtk4
+  ];
+
+  configureFlags = [
+    "--with-gtk4"
+    "--enable-absolute-paths"
+  ];
+
+  strictDeps = true;
+
+  passthru = {
+    networkManagerPlugin = "VPN/nm-ssh-service.name";
+  };
+
+  meta = with lib; {
+    description = "SSH VPN integration for NetworkManager";
+    license = licenses.gpl2Plus;
+  };
```

1. `with` before attrsets (especially `with lib;`, _especially_ file-wide ones, _**especially**_ multiple ones) has been abused to death in the past and is now frowned upon[^1]. Just prepend `lib.` where necessary
2. Really poor meta, needs more fields to be set (homepage, platforms, you as a maintainer[^2])
```suggestion
  meta = {
    description = "SSH VPN integration for NetworkManager";
    homepage = "https://github.com/danfruehauf/NetworkManager-ssh";
    license = licenses.gpl2Plus;
    inherit (networkmanager.meta) platforms;
  };
```

[^1]: `with` before small scopes, like lists, is fine
[^2]: you must add yourself (optionally with other willing folks) as a maintainer of the initiated package. maintainer entry should be added in a separate commit (within this PR! don't open a new one) that _precedes_ newly added content